external-crates/move/move-compiler/src/parser/syntax.rs

```diff
@@ -2104,6 +2126,63 @@ fn parse_field_annot(context: &mut Context) -> Result<(Field, Type), Box<Diagnos
     Ok((f, st))
 }
 
+// Parse a postfix ability declaration:
+//     "has" <Ability> (, <Ability>)+;
+//  Error if:
+//      * Also has prefix ability declaration
+fn parse_postfix_ability_declarations(
+    infix_ability_declaration_loc: Option<Loc>,
+    abilities: &mut Vec<Ability>,
+    context: &mut Context,
+) -> Result<(), Box<Diagnostic>> {
+    let postfix_ability_declaration =
+        context.tokens.peek() == Tok::Identifier && context.tokens.content() == "has";
+    let has_location = current_token_loc(context.tokens);
+
+    if postfix_ability_declaration
+        && context.env.check_feature(
```

Comment:
Casing on this check (and converting `check_feature` to return a `bool` seems unnecessary. You can parse this and let the `check_feature` call produce the relevant error. This will allow compilation to continue with that error in place.

If you'd prefer to keep it this way, IMO this should have an else case with an explicit error.

Reply:
Yeah, I agree with @cgswords in this case it it might be best to produce an explicit error and still parse the `has`.

Though we will need to produce the bool for `public struct` since that is a bit of a different change.